azure/services/aso/service.go

```diff
@@ -62,18 +66,17 @@ func (s *Service[T, S]) Reconcile(ctx context.Context) error {
 	ctx, cancel := context.WithTimeout(ctx, s.Scope.DefaultedAzureServiceReconcileTimeout())
 	defer cancel()
 
-	if len(s.Specs) == 0 {
-		return nil
-	}
```

Comment:
This is to make sure we still clean up resources if all references were removed from a CAPZ spec. This essentially reverts #4324 so it might muddy some of the status reporting, so please try these changes for yourself to see if any of the status conditions seem confusing for what's actually happening while a resource is being deleted.